revup/topic_stack.py

```diff
@@ -179,6 +179,21 @@ class Topic:
     # Reviews for this topic, keyed by base branch
     reviews: Dict[str, Review] = field(default_factory=dict)
 
+    def depth_first_iter(self) -> Iterator[Topic]:
+        for t in self.depth_first_iter_helper(set()):
+            yield t
+
+    def depth_first_iter_helper(self, seen: Set[str]) -> Iterator[Topic]:
```

Comment:
Maybe this should be private, since you probably shouldn't call it directly?  Or it could even be a local function inside `depth_first_iter` I think?